CMakeLists.txt

```diff
@@ -61,6 +61,7 @@ option (ENABLE_PIC
    "Enables building of position independent code for static library components."
    ON
 )
+option (ENABLE_TEST_ASSERTIONS "Turn on mutex lock asserts" OFF)
```

Comment:
Thanks for renaming this. At the risk of continued bikeshedding, can you rename this to `ENABLE_DEBUG_ASSERTIONS`? I think that conveys better that these are assertions intended for debugging only. And can you reword the help description to be more generic, like "Turn on runtime test assertions for debugging"?